chcemvediet/apps/inforequests/admin.py

```diff
@@ -385,20 +389,69 @@ def get_inforequest(self, obj):
         return obj.branch.inforequest
 
     def delete_constraints(self, obj):
-        dependency = []
+        constraints = []
         if obj.type in [Action.TYPES.REQUEST, Action.TYPES.ADVANCED_REQUEST]:
-            dependency.append(format_html(
+            constraints.append(format_html(
                 u'{} is type {}.'.format(admin_obj_format(obj), obj.get_type_display())))
         if len(obj.branch.actions) == 1:
-            dependency.append(format_html(
+            constraints.append(format_html(
                 u'{} is the only action in the branch.'.format(admin_obj_format(obj))))
-        return dependency
+        return constraints
 
     def render_delete_form(self, request, context):
         context[u'delete_constraints'] = self.delete_constraints(context[u'object'])
         context[u'ADMIN_EXTEND_SNOOZE_BY_DAYS'] = ADMIN_EXTEND_SNOOZE_BY_DAYS
         return super(ActionAdmin, self).render_delete_form(request, context)
 
+    @transaction.atomic
+    def delete_selected(self, request, queryset):
+        constraints = []
+        warnings = []
+        outbound = InforequestEmail.objects.none()
+        inbound = InforequestEmail.objects.none()
+        for obj in queryset:
+            obj_constraints = self.delete_constraints(obj)
+            constraints += obj_constraints
+            inforequestemails = self.nested_inforequestemail_queryset(obj)
+            outbound |= inforequestemails[0]
+            inbound |= inforequestemails[1]
+            if not obj_constraints:
+                if not obj.is_last_action and obj.next_action not in queryset:
+                    warnings.append(format_html(squeeze(u"""
+                            {} is not the last action in the branch. Deleting it may cause logical
+                            errors in the inforequest history.
+                            """).format(admin_obj_format(obj))))
+
+        if request.POST.get(u'post'):
```

Comment:
@martinmacko47 rozdelil som to na 2x POST, aby sa najprv vykonala najprv nasa kontrola, potom aby sa vykonala kontrola od django a az potom nastane nami pridana uprava outbound a inbound.

Reply:
@viliambalaz To, ze je 2x kontrola na POST nevadi. Nechapem vsak, preco pises:

```
if request.POST.get(u'post'):
```

Tato podmienka predsa neznamena, ze request je POST, ale ze je to POST a zaroven z FE sa poslal field s nazvom `post`. Nemala by ta podmienka byt iba takto?

```
if request.POST:
```

Reply:
Je to z toho dovodu, ze template `admin/delete_selected_confirmation.html` ma v sebe dodatocny `<input name=post`
```
<form action="" method="post">{% csrf_token %}
    <div>
    ...
    <input type="hidden" name="post" value="yes" />
    <input type="submit" value="{% trans "Yes, I'm sure" %}" />
    </div>
</form>
```
Zrejme to je z toho dovodu, aby bolo mozne pridat viacero roznych `POST` requestov na danu url. Takto to je spravene v celom `django` na jedinom mieste, a to je v admine pre delete_selected. Ja by som ostal pri pouzivani `if request.POST.get(u'post')`

Reply:
Je to z toho dovodu, aby bolo mozne vykonat 2 rozne POST metody na url `/en/admin/inforequests/action/`. Prva metoda sa uskutocnuje cez button **Go**, druha cez **Yes I'm sure**
![Screenshot_2021-05-17 Select action to change Django site admin](https://user-images.githubusercontent.com/29918752/118482610-41bc7c80-b715-11eb-81fe-356084a9e56f.png)
Ak by to nebolo takto riesene, po kliknuti na **Go** by sa nam neukazala potvrdzovacia stranka, nastalo by hned vymazanie objektov.

Reply:
Chapem. V `delete_model` metode tento problem nenastava? Tam moze ostat `if request.POST:`?

Reply:
Nie v `delete_model` (* `delete_view`) tenot problem nenastava. Tlacidlo delete na obrazovke detaile Action sa vykona GET metoda na `/admin/inforequests/action/<pk>/delete/`, samotne mazanie prebieha s metodou POST na rovnaku url.